siliconcompiler/tools/yosys/yosys.py

```diff
@@ -402,15 +402,15 @@ def get_synthesis_corner(chip):
 
     # determine corner based on setup corner from constraints
     corner = None
-    for constraint in chip.getkeys('constraint'):
-        if "setup" in chip.get('constraint', constraint, 'check') and not corner:
-            corner = chip.get('constraint', constraint, 'libcorner')
+    for constraint in chip.getkeys('constraint', 'timing'):
```

Comment:
should we check if `'constraint','timing'` is valid first?

Reply:
Actually, should not be needed since timing is always defined.